src/srv.rs

```diff
@@ -39,7 +37,10 @@ impl SrvResolver {
         Ok(config)
     }
 
-    async fn get_srv_hosts(&self, original_hostname: &str) -> Result<Vec<StreamAddress>> {
+    pub(crate) async fn get_srv_hosts<'a>(
```

Comment:
In order to properly implement the SRV polling spec, we need to keep the valid subset of records even if some of them are malformed, which differs from the initial seedlist spec, where a single error is allowed to cause the entire lookup to be considered failed. In order to facilitate this, I've changed the return value to iterate over `Result<StreamAddress>` instead of just shifting all of the internally errors to the outside `Result`.